__test__/(not-authenticated)/sign_in.test.jsx

```diff
@@ -0,0 +1,84 @@
+import { render, screen } from '@testing-library/react';
+import SignIn from '../../src/app/(not-authenticated)/sign_in/page';
+import '@testing-library/jest-dom';
+
+// mock useRouter
+jest.mock('next/navigation', () => ({
+  useRouter: jest.fn(),
+}));
+
+const pushMock = jest.fn();
+jest.mock('next/navigation', () => ({
+  useRouter() {
+    return {
+      query: {},
+      push: pushMock,
+      pathname: '',
+      asPath: '',
+      events: {
+        on: jest.fn(),
+        off: jest.fn(),
+      },
+    };
+  },
+}));
+
+describe('Sign In', () => {
+  it('renders a heading', () => {
+    render(<SignIn />);
+
+    const heading = screen.getByRole('heading', {
+      name: 'Black Market',
+    });
```

Comment:
minor thing: you can also add the `level` option to make more specific
```suggestion
    const heading = screen.getByRole('heading', {
      name: 'Black Market',
      level: 1
    });
```